### ISSUE TITLE
No warning for implicit narrowing conversions to enums in C.

### ISSUE THREAD

**Issue by tstanisl** [llvm/llvm-project#65727]
No warning is raised for implicit conversions to lower precision enumeration types.

```
enum Color { RED, GREEN, BLUE } e = 3.14;
int i = 3.14;
```
When compiled with `-c -Wall -Wextra -pedantic` options the only warning refers to implicit `double` to `int` conversion.
There is no warning about conversion from `double` to `enum Color`.
```
<source>:2:9: warning: implicit conversion from 'double' to 'int' changes value from 3.14 to 3 [-Wliteral-conversion]
int i = 3.14;
```

See [godbolt](https://godbolt.org/z/fo4qeTcq5).

**Comment by shafik** [comment]
It looks like we only do `DiagnoseFloatingImpCast(...)` for builtin types:

https://github.com/llvm/llvm-project/blob/f2b04434fc49e41444d1a1a4065e8fbe249538c2/clang/lib/Sema/SemaChecking.cpp#L14976-L14982

Not sure what would happen if we expanded this there would be issues or not. 

CC @AaronBallman since this is C

**Comment by AaronBallman** [comment]
I think it would be reasonable for us to expand that to cover enumeration types as well. FWIW, ICC warns on it, GCC does not.
